service/src/main/kotlin/fi/espoo/evaka/placement/PlacementPlanQueries.kt

```diff
@@ -259,44 +259,47 @@ fun Database.Read.getGuardiansRestrictedStatus(guardianId: PersonId): Boolean? {
         .singleOrNull()
 }
 
-fun Database.Transaction.terminatePlacementFrom(terminationRequestedDate: LocalDate, placementId: PlacementId, placementTerminationDate: LocalDate, terminatedBy: EvakaUserId?) {
+fun Database.Transaction.deleteServiceNeedsFromPlacementAfter(id: PlacementId, date: LocalDate) {
     createUpdate(
-        //language=SQL
         """
-UPDATE daycare_group_placement 
-SET end_date = :placementTerminationDate 
-WHERE daycare_placement_id = :placementId
+DELETE FROM service_need 
+WHERE placement_id = :placementId
+    AND start_date > :date
         """.trimIndent()
     )
-        .bind("placementId", placementId)
-        .bind("placementTerminationDate", placementTerminationDate)
+        .bind("placementId", id)
+        .bind("date", date)
         .execute()
 
     createUpdate(
         //language=SQL
         """
-DELETE FROM service_need 
+UPDATE service_need 
+SET end_date = :date 
 WHERE placement_id = :placementId
-    AND start_date > :placementTerminationDate
+    AND daterange(start_date, end_date, '[]') @> :date::date
         """.trimIndent()
     )
-        .bind("placementId", placementId)
-        .bind("placementTerminationDate", placementTerminationDate)
+        .bind("placementId", id)
+        .bind("date", date)
         .execute()
+}
 
+fun Database.Transaction.terminatePlacementFrom(terminationRequestedDate: LocalDate, placementId: PlacementId, placementTerminationDate: LocalDate, terminatedBy: EvakaUserId?) {
     createUpdate(
         //language=SQL
         """
-UPDATE service_need 
+UPDATE daycare_group_placement 
 SET end_date = :placementTerminationDate 
-WHERE placement_id = :placementId
-    AND daterange(start_date, end_date, '[]') @> :placementTerminationDate::date
+WHERE daycare_placement_id = :placementId
```

Comment:
What about placements with multiple group placements? Are those handled somewhere else?

Reply:
Right. I guess those should be handled similarly to service needs.

Reply:
Fixed this as well by re-using logic. These queries were not related to placement plans so I moved them to `PlacementQueries.kt`.